internal/server/singleprocess/state/instance_exec.go

```diff
@@ -77,6 +107,13 @@ func (s *State) InstanceExecCreateByDeployment(did string, exec *InstanceExec) e
 	for raw := iter.Next(); raw != nil; raw = iter.Next() {
 		rec := raw.(*Instance)
 
+		// When looking through all the instances for an exec capable instance
+		// we only consider LONG_RUNNING type instances. These are the only ones
+		// that it makes sense to send random exec sessions to.
+		if rec.Type != pb.Instance_LONG_RUNNING {
```

Comment:
Can you add a unit test for this functionality?

Reply:
Ah! I missed that one, yup.